src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx

```diff
@@ -396,6 +416,85 @@ export class DashboardContainer
     return newId;
   }
 
+  public async addNewPanel<ApiType extends unknown = unknown>(
+    panelPackage: PanelPackage,
+    silent?: boolean
+  ) {
+    if (reactEmbeddableRegistryHasKey(panelPackage.panelType)) {
+      const newId = v4();
+      const { newPanelPlacement, otherPanels } = panelPlacementStrategies.findTopLeftMostOpenSpace({
+        currentPanels: this.getInput().panels,
+        height: DEFAULT_PANEL_HEIGHT,
+        width: DEFAULT_PANEL_WIDTH,
+      });
+      const newPanel: DashboardPanelState = {
+        type: panelPackage.panelType,
+        gridData: {
+          ...newPanelPlacement,
+          i: newId,
+        },
+        explicitInput: {
+          ...panelPackage.initialState,
+          id: newId,
+        },
+      };
+      this.updateInput({ panels: { ...otherPanels, [newId]: newPanel } });
+      return;
+    }
+
+    const {
+      notifications: { toasts },
+      embeddable: { getEmbeddableFactory },
+    } = pluginServices.getServices();
+    const embeddableFactory = getEmbeddableFactory(panelPackage.panelType);
+    if (!embeddableFactory) {
+      throw new EmbeddableFactoryNotFoundError(panelPackage.panelType);
+    }
+
+    if (this.trackPanelAddMetric) {
```

Comment:
Should trackPanelAddMetric be moved to top of function so it can track react embeddables as well?